caampr-backend/src/main/java/com/jelistan/caampr/lambda/dao/GearDao.java

```diff
@@ -0,0 +1,85 @@
+package com.jelistan.caampr.lambda.dao;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.ComparisonOperator;
+import com.amazonaws.services.dynamodbv2.model.Condition;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.jelistan.caampr.lambda.model.Gear;
+import com.jelistan.caampr.lambda.model.GearListRequest;
+import com.jelistan.caampr.lambda.model.VisibilityTypes;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Data Access Object for the Gear table
+ */
+public class GearDao {
+
+    private final DynamoDBMapper mapper;
+
+    /**
+     * Constructor for Gear Provider implementation, takes Dynamo mapper as input.
+     * @param mapper
+     */
+    @Inject
+    public GearDao(final DynamoDBMapper mapper){
+        this.mapper = mapper;
+    }
+
+    /**
+     * Fetches a JSON list of gear objects from the Dynamo server based on the gears' associated profile ID, and
+     * deserializes it into a List<Gear>.
+     * @param request The GearListRequest object containing the required info (profileId, etc)
+     * @return The List of Gear objects with matching profile IDs, which may be an empty list.
+     */
+    public List<Gear> getList(final GearListRequest request){
```

Comment:
Is differentiation based on list entry type (GEAR or VEHICLE) going to be added later as a parameter? Otherwise I dont see why this should be refactored away from getGearList

Reply:
Oh, i see, its a GEAR dao, not a general dao. Nvm. Different question then, it seems each DAO will have significant overlap, so is it planned to have some form of inheritance or something between them?

Reply:
this is the DAO for the gear table, so by type.  The input request has the profileId and type set and that's what we will query against.  If we need to add VEHICLE, we can use the same `getList` method without touching anything.